keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -293,33 +291,49 @@ def _augment(self, inputs):
         return result
 
     def _batch_augment(self, inputs):
-        return self._map_fn(self._augment, inputs)
+        augmented_batch = self._map_fn(self._augment, inputs)
+        return augmented_batch
 
     def _format_inputs(self, inputs):
         if tf.is_tensor(inputs):
             # single image input tensor
-            return {IMAGES: inputs}, False, False
+            meta_data = {"is_dict": False, "use_targets": False}
+            inputs = {IMAGES: inputs}
         elif isinstance(inputs, dict) and TARGETS in inputs:
             # TODO(scottzhu): Check if it only contains the valid keys
             inputs[LABELS] = inputs[TARGETS]
             del inputs[TARGETS]
-            return inputs, True, True
+            meta_data = {"is_dict": True, "use_targets": True}
         elif isinstance(inputs, dict):
-            return inputs, True, False
+            meta_data = {"is_dict": True, "use_targets": False}
         else:
             raise ValueError(
                 f"Expect the inputs to be image tensor or dict. Got {inputs}"
             )
+        if BOUNDING_BOXES in inputs:
+            if isinstance(inputs[BOUNDING_BOXES], tf.RaggedTensor):
+                meta_data["is_bounding_boxes_ragged"] = True
+                meta_data["ragged_row_lengths"] = inputs[
+                    BOUNDING_BOXES
+                ].nested_row_lengths()
```

Comment:
one caveat to this approach: what if the number of bounding boxes changes in the layer?  For example, RandomCrop() will remove some bounding boxes

This may be a point in favor of shipping a utility to go from padded Tensor -> ragged

Reply:
Do you think adding a zero area filter would be good? And augmentations never drop any boxes, they would just get clipped to 0?

Reply:
done!